From fd88296ca4eb666b2fa80caf1f9e70f162f2ffba Mon Sep 17 00:00:00 2001
From: Pradeep Kumar Chitrapu <quic_pradeepc@quicinc.com>
Date: Wed, 10 Jan 2024 19:28:04 -0800
Subject: [PATCH 2/2] wifi: ath12k: In fast Rx path, do not submit frames to
 PPE for RFS

This patch ensures that frames are not delivered to PPE when RFS
is enabled. With RFS enabled, PPE only handles steering the
flows across CPU cores and does not involve in packet editing.

Signed-off-by: Pradeep Kumar Chitrapu <quic_pradeepc@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/peer.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/net/wireless/ath/ath12k/peer.c b/drivers/net/wireless/ath/ath12k/peer.c
index fc17fbf626b5..89f5ca2cf87f 100644
--- a/drivers/net/wireless/ath/ath12k/peer.c
+++ b/drivers/net/wireless/ath/ath12k/peer.c
@@ -705,7 +705,11 @@ int ath12k_peer_create(struct ath12k *ar, struct ath12k_link_vif *arvif,
 	peer->sec_type_grp = HAL_ENCRYPT_TYPE_OPEN;
 	peer->vif = vif;
 #ifdef CPTCFG_MAC80211_PPE_SUPPORT
-	peer->ppe_vp_num = peer->vif->ppe_vp_num;
+	/* Do not deliver frames to PPE in fast rx incase of RFS
+	 * RFS is supported only in SFE Mode
+	 */
+	if (peer->vif->ppe_vp_type == PPE_VP_USER_TYPE_ACTIVE)
+		peer->ppe_vp_num = peer->vif->ppe_vp_num;
 #endif
 #ifdef CPTCFG_ATH12K_BONDED_DS_SUPPORT
 	peer->ppe_vp_num = arvif->ppe_vp_num;
-- 
2.17.1

